frontend/heir/backends/util/clang.py

```diff
@@ -30,6 +30,46 @@ def _find_clang(self):
       return clang
     return shutil.which("clang")
 
+
+  def clang_arg_helper(
```

Comment:
This is currently only used in debug mode. Could you also use it in the actual compilation func below, so that we avoid code drift between the two implementations?

You could also just pass a debug mode (or a debug callback) to the clang helper and have it print the args there.